qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,335 @@
+            self._tmp_ansatz.operators = self._excitation_list + [exc]
+            # the ansatz needs to be decomposed for the gradient to work
+            self.solver.ansatz = self._tmp_ansatz.decompose()
+            param_sets = list(self.solver.ansatz.parameters)
+            # zip will only iterate the length of the shorter list
+            theta1 = dict(zip(self.solver.ansatz.parameters, theta))
+            op, expectation = self.solver.construct_expectation(
+                theta1, operator, return_expectation=True
+            )
+            # compute gradient
+            state_grad = self._adapt_gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            value_dict = {param_sets[-1]: 0.0}
+            # print(state_grad)
+            state_grad_result = sampler.convert(state_grad, params=value_dict).eval()
+            logger.info("Gradient computed : %s", str(state_grad_result))
+            res.append((np.abs(state_grad_result[-1]), exc))
+        return res, expectation
+
+    @staticmethod
+    def _check_cyclicity(indices: List[int]) -> bool:
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: The list of chosen gradient indices.
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        # reg-ex explanation:
+        # 1. (\b.+ .+\b) will match at least two numbers and try to match as many as possible. The
+        #    word boundaries in the beginning and end ensure that now numbers are split into digits.
+        # 2. the match of this part is placed into capture group 1
+        # 3. ( \b\1\b)+ will match a space followed by the contents of capture group 1 (again
+        #    delimited by word boundaries to avoid separation into digits).
+        # -> this results in any sequence of at least two numbers being detected
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    @staticmethod
+    def _log_gradients(
+        iteration: int,
+        cur_grads: Tuple[float, PauliSumOp],
+        max_grad: Tuple[float, PauliSumOp | None],
+    ):
+        if logger.isEnabledFor(logging.INFO):
+            gradlog = f"\nGradients in iteration #{str(iteration)}"
+            gradlog += "\nID: Excitation Operator: Gradient  <(*) maximum>"
```

Comment:
Can it be 1 assignment?

Reply:
No cause then I am getting line too long error when I run "make lint"

Reply:
That should not force us to having the code implemented this way. If the black does not format the too long line automatically, you might break it manually but still having a single assignment to a variable.